modules/grpc/common/grpc-source.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2024 Axoflow
+ * Copyright (c) 2023-2024 Attila Szakacs <attila.szakacs@axoflow.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "grpc-source.hpp"
+
+#include <string>
+
+using namespace syslogng::grpc;
+
+/* C++ Implementations */
+
+SourceDriver::SourceDriver(GrpcSourceDriver *s)
+  : super(s)
+{
+  credentials_builder_wrapper.self = &credentials_builder;
+}
+
+gboolean
+SourceDriver::init()
+{
+  if (!this->port)
+    {
+      msg_error("Failed to initialize gRPC based source, port() must be set",
+                log_pipe_location_tag(&this->super->super.super.super.super));
+      return FALSE;
+    }
+
+  ::grpc::EnableDefaultHealthCheckService(true);
+
+  this->super->super.worker_options.super.init_window_size /= super->super.num_workers;
```

Comment:
This is handled by threaded source now.

We should move this logic here in order not to revert #310 accidenteally:
```
 if (fetch_limit == -1)
    {
      if (super->super.worker_options.super.init_window_size != -1)
        fetch_limit = super->super.worker_options.super.init_window_size / super->super.num_workers;
      else
        fetch_limit = 100;
    }
```

Reply:
nice catch!

squashed the fixup commits and fixed `grpc::SourceDriver::init()` in its original commit,